### ISSUE TITLE
`test_psd_matlab` fails with matplotlib 3.6.2

### ISSUE THREAD

**Issue by risicle** [nipy/nitime#198]
Along the lines of #197, when we bumped the NixOS package for `matplotlib` to 3.6.2, `test_psd_matlab` in nitime 0.9 started failing:

```
_______________________________ test_psd_matlab ________________________________

    def test_psd_matlab():
    
        """ Test the results of mlab csd/psd against saved results from Matlab"""
    
        from matplotlib import mlab
    
        test_dir_path = os.path.join(nitime.__path__[0], 'tests')
    
        ts = np.loadtxt(os.path.join(test_dir_path, 'tseries12.txt'))
    
        #Complex signal!
        ts0 = ts[1] + ts[0] * np.complex(0, 1)
    
        NFFT = 256
        Fs = 1.0
        noverlap = NFFT / 2
    
>       fxx, f = mlab.psd(ts0, NFFT=NFFT, Fs=Fs, noverlap=noverlap,
                          scale_by_freq=True)

nitime/tests/test_algorithms.py:154: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/nix/store/fq03dka9ivs40a4hjn0aajl7m5az48g0-python3.10-matplotlib-3.6.2/lib/python3.10/site-packages/matplotlib/mlab.py:584: in psd
    Pxx, freqs = csd(x=x, y=None, NFFT=NFFT, Fs=Fs, detrend=detrend,
/nix/store/fq03dka9ivs40a4hjn0aajl7m5az48g0-python3.10-matplotlib-3.6.2/lib/python3.10/site-packages/matplotlib/mlab.py:640: in csd
    Pxy, freqs, _ = _spectral_helper(x=x, y=y, NFFT=NFFT, Fs=Fs,
/nix/store/fq03dka9ivs40a4hjn0aajl7m5az48g0-python3.10-matplotlib-3.6.2/lib/python3.10/site-packages/matplotlib/mlab.py:382: in _spectral_helper
    result = _stride_windows(x, NFFT, noverlap)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

x = array([ 8.8313817e+00+8.9429666e+00j,  4.3005055e+00+5.0100717e+00j,
       -4.1649119e+00-2.6816087e+00j, -6.3050160e...01-3.9449765e+00j,  4.4173569e+00+1.4387487e+00j,
        7.1039609e+00+1.0514320e+01j,  5.4811765e+00+1.0769771e+01j])
n = 256, noverlap = 128.0, axis = 0

    def _stride_windows(x, n, noverlap=0, axis=0):
        # np>=1.20 provides sliding_window_view, and we only ever use axis=0.
        if hasattr(np.lib.stride_tricks, "sliding_window_view") and axis == 0:
            if noverlap >= n:
                raise ValueError('noverlap must be less than n')
>           return np.lib.stride_tricks.sliding_window_view(
                x, n, axis=0)[::n - noverlap].T
E           TypeError: slice indices must be integers or None or have an __index__ method
```

Looks like `noverlap` now just needs to be cast to an integer before being passed to `mlab.psd`.

**Comment by arokem** [comment]
Do you want to put in a PR fixing this?

**Comment by effigies** [comment]
Fixed in #200.
